src/monitor/src/main/java/ee/ria/xroad/monitor/MetricRegistryHolder.java

```diff
@@ -22,13 +22,19 @@
  */
 package ee.ria.xroad.monitor;
 
+import com.codahale.metrics.Histogram;
 import com.codahale.metrics.MetricRegistry;
+import com.codahale.metrics.SlidingTimeWindowReservoir;
+
+import java.io.Serializable;
+import java.util.concurrent.TimeUnit;
 
 /**
  * Global access point for {@link MetricRegistry}
```

Comment:
getOrCreateXXX methods are not thread safe, if multiple concurrent threads use them with same metricName, dangling SimpleSensors/Histogram objects are created. Either add comments about thread safety, or implement locking to achieve thread safety.

Reply:
Now they are thread safe!